arrow/parser.py

```diff
@@ -120,6 +122,7 @@ def parse_iso(self, datetime_string):
         has_tz = False
 
         # TODO: add tests for all the new formats, especially basic format
+        # NOTE: YYYYMM is omitted to avoid confusion with YYMMDD (no longer part of ISO 8601, but is still often used)
```

Comment:
We don't test for `YYMMDD` in `parse_iso` though so `YYYYMM` could be allowed?

Reply:
It could be, but I am worried about ambiguity. I think we should require a format string for ambiguous date formats that we are not 100% sure about. We don't want people complaining about `get` being erratic anymore :D.

Reply:
so for example someone may pass `"200806"` expecting `2020-08-06` but end up with `2008-06-01` ?

Reply:
Yup, exactly!

Also, I am reading the ISO 8601 Wikipedia page and it says the following:  "By disallowing dates of the form YYYYMM, the standard avoids confusion with the truncated representation YYMMDD (still often used)."

It seems to be explicitly disallowed by the standard, so maybe I should change the comment to reflect that:
<img width="293" alt="Screen Shot 2019-08-19 at 5 59 40 PM" src="https://user-images.githubusercontent.com/7388203/63275987-2fe14780-c2ab-11e9-9f78-de57498c8fb8.png">

Reply:
Ah ok then fair enough let's avoid trying `YYYYMM`.